### ISSUE TITLE
Azure SQL Server: reading view definition, procedure 'sp_helptext' is not supported

### ISSUE THREAD

**Issue by dr-1** [dbeaver/dbeaver#15490]
#### System information: 
- Windows 10
- DBeaver 21.3.4

#### Connection specification:
- Database type: Azure SQL Server
- Driver name: Azure SQL Server
- No tunnels or proxies

#### Describe the problem you're observing:
Trying to get the DDL of a view produces this error:
```
/* ERROR WHILE READING SOURCE:

SQL Error [15871] [S0004]: PROCEDURE 'sp_helptext' is not supported.
*/
```

#### Steps to reproduce, if exist:
- Open a view object in this type of database
- Go to Properties -> Definition

#### Include any warning/errors/backtraces from the logs
eclipse.buildId=unknown
java.version=11.0.12
java.vendor=Eclipse Foundation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en_US
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Mon Feb 14 17:31:21 CET 2022
SQL Error [15871] [S0004]: PROCEDURE 'sp_helptext' is not supported.

This issue, previously reported in #6651 and #6979, was fixed for some cases in PR #7842 which introduced a data source check to apply an alternative way to read definitions. In my case, this check seems to return false and consequently the query using the unsupported sp_helptext is run.

**Comment by LonwoLonwo** [comment]
Hello @dr-1 

What Azure version do you use? Maybe do you have any ideas - why the support of 'sp_helptext' is not supported in your Azure version?

**Comment by dr-1** [comment]
Hi @LonwoLonwo, it seems that Azure does not provide version information, at least I cannot find any reference to that in its portal and documentation.

What I know is that the database is hosted in an instance of the Synapse Analytics service, which does not support sp_helptext according to MS' [docs](https://docs.microsoft.com/en-us/sql/relational-databases/system-stored-procedures/sp-helptext-transact-sql) (does not say why), and is of the "serverless" type.

**Comment by LonwoLonwo** [comment]
Thanks for the answer @dr-1 

You can check your database/driver version in the Connection Settings -> Test Connection.

![2022-02-16 11_31_00-DBeaver Ultimate 21 3 5 - TEST_COMPLEX_UNIQUE](https://user-images.githubusercontent.com/45152336/154225939-532c6d58-d0a9-4528-a24f-0e0a845587f8.png)

Also, can you say me please - what returns you this statement:
`SELECT SERVERPROPERTY('EngineEdition')`

**Comment by dr-1** [comment]
Thanks for those hints, @LonwoLonwo. The connection test shows these versions:

Server:
Microsoft SQL Server 12.00.63
Microsoft SQL Azure (RTM) - 12.0.2000.8 
	Jan 27 2022 07:48:10 
	Copyright (C) 2021 Microsoft Corporation

Driver:
Microsoft JDBC Driver 9.2 for SQL Server 9.2.0.0

`SELECT SERVERPROPERTY('EngineEdition')` returns 11.

**Comment by LonwoLonwo** [comment]
Ok. 
As you said, it was "fixed for some cases in PR https://github.com/dbeaver/dbeaver/pull/7842"

So, please tell me, do this workaround statement (from PR) work for you?

`SELECT definition FROM sys.sql_modules WHERE object_id = (OBJECT_ID(N'your_shema_name.your_procedure_name'))`

**Comment by dr-1** [comment]
Yes, this works, returning the expected definition.

**Comment by LonwoLonwo** [comment]
Thanks for the investigation @dr-1

**Comment by LonwoLonwo** [comment]
Fixed in DBeaver 22.0.0 version

**Comment by Matvey16** [comment]
Verified
